whylog/teacher/constraint_links_base.py

```diff
@@ -0,0 +1,48 @@
+class ConstraintLinksBase:
+    """
+    Container for links between group and constraint
+    """
+
+    def __init__(self, links_list=None):
+        """
+        :param links_list: list of links between group and constraint
+                          one link is a tuple (line_id, group_no, constraint_id)
+        :type links_list: list[(int, int, int)]
+        """
+        self.links = links_list or []
+        self.constr_id_position = 2
+
+    def get_links(self):
+        return self.links
+
+    def distinct_constraint_ids(self):
+        return set(list(zip(*self.links))[self.constr_id_position])
+
+    def insert_links(self, links_list):
+        self.links = list(set(self.links + links_list))
```

Comment:
So what this code do? Convert concatenation of two lists into set and then convert into list. The result is the same as sort if the input is different. For duplicates, the code delete one of them. I think in this situations code should rather raise exception than delete one of them.